src/CommandIsolator.cpp

```diff
@@ -82,13 +92,37 @@ CommandIsolatorProcess::CommandIsolatorProcess(
       m_usageCommand(usageCommand),
       m_isDebugMode(isDebugMode) {}
 
+Try<Nothing> CommandIsolatorProcess::saveContainerContext(const ContainerID& containerId, const ContainerConfig& containerConfig) {
+    const string& context_file_path = path::join(COMMAND_ISOLATOR_STATE_DIR, stringify(containerId));
+    Result<Nothing> ok = os::write(context_file_path, stringify(JSON::protobuf(containerConfig)));
+    if (ok.isError()) {
+        return Error("Failed writing context file for container " + stringify(containerId) + ": " + ok.error());
+    }
+    return Nothing();
+}
+
+Try<ContainerConfig> CommandIsolatorProcess::restoreContainerContext(const ContainerID& containerId) {
+    const string& context_file_path = path::join(COMMAND_ISOLATOR_STATE_DIR, stringify(containerId));
+    Result<string> context_json = os::read(context_file_path);
+    if (context_json.isError()) {
+        return Error("Failed reading context file: " + context_json.error());
+    }
+    Result<ContainerConfig> containerConfig = 
+            jsonToProtobuf<ContainerConfig>(context_json.get());
+    if (containerConfig.isError()) {
+        return Error("Unable to deserialize ContainerConfig: " + containerConfig.error());
+    }
+    return containerConfig.get();
+}
+
 process::Future<Option<ContainerLaunchInfo>> CommandIsolatorProcess::prepare(
     const ContainerID& containerId, const ContainerConfig& containerConfig) {
   if (m_infos.contains(containerId)) {
     return Failure("mesos-command-module already initialized for container");
   } else {
     m_infos.put(containerId, containerConfig);
   }
+  saveContainerContext(containerId, containerConfig);
```

Comment:
Should we try to do something if we fail here? (open question, I have no idea how we can recover from this :( )

Reply:
Good point. I think the discussion is open. I'm not sure we'd like to ensure recovery to the point of preventing a task from launching just because we couldn't save its context on disk. But on the other hand, if some bug is massively happening, you'd like to get a clear signal... Let's discuss this point with the others.